src/org/mozilla/classfile/ClassFileWriter.java

```diff
@@ -1605,7 +1605,11 @@ private void killSuperBlock(SuperBlock sb) {
                     int handlerPC = getLabelPC(ete.itsHandlerLabel);
                     SuperBlock handlerSB = getSuperBlockFromOffset(handlerPC);
                     locals = handlerSB.getLocals();
-                    break;
+                    if (handlerSB.isInitialized()) {
+                        break;
+                    } else {
+                        continue;
```

Comment:
Do you need to `continue`?
Since it is `if (sbStart > eteStart &&` and ` if (eteStart > sbStart &&`, the result is the same, but it feels more readable to write the same code as here.
https://github.com/mozilla/rhino/blob/f33a0c11f24f86fc6fff433f632a5013ca50e9af/src/org/mozilla/classfile/ClassFileWriter.java#L1617-L1620

Reply:
Thank you for your review. I try to be efficient in a hot code path (the comment immediately above says so, and I agree), and while perhaps a smart JIT compiler would put a jump right there, I'd rather put it myself and make no assumptions about the JIT. :-)